# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1451883891 18000
#      Mon Jan 04 00:04:51 2016 -0500
# Node ID 13b9cac748f2d3c0f2c1808d8dffcd31a74f285b
# Parent  66a08c3cafe3a56dc0e27c85c4747e43a56a935e
avoid HDF5 macros if HDF5 is not available

* oct-hdf5.h: Don't declare or define anything unless HAVE_HDF5
is defined.

* coct-hdf5-types.c, oct-hdf5-types.cc: Don't include oct-hdf5-types.h
seperately.
* coct-hdf5-types.c: Don't define HDF5 constants unless HAVE_HDF5 is
defined.

diff --git a/libinterp/corefcn/coct-hdf5-types.c b/libinterp/corefcn/coct-hdf5-types.c
--- a/libinterp/corefcn/coct-hdf5-types.c
+++ b/libinterp/corefcn/coct-hdf5-types.c
@@ -20,20 +20,23 @@ along with Octave; see the file COPYING.
 
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include "oct-hdf5.h"
-#include "oct-hdf5-types.h"
 
 // Programming Note: This file exists so that we can hide the use
 // of macros and C-style casts in a C warnings about using old-style
 // casts in C++.
 
 // Please do NOT eliminate this file and move code from here to
 // oct-hdf5-types.cc
 
+#if defined (HAVE_HDF5)
+
 const octave_hdf5_id octave_H5E_DEFAULT = H5E_DEFAULT;
 const octave_hdf5_id octave_H5P_DEFAULT = H5P_DEFAULT;
 const octave_hdf5_id octave_H5S_ALL = H5S_ALL;
+
+#endif
diff --git a/libinterp/corefcn/oct-hdf5-types.cc b/libinterp/corefcn/oct-hdf5-types.cc
--- a/libinterp/corefcn/oct-hdf5-types.cc
+++ b/libinterp/corefcn/oct-hdf5-types.cc
@@ -21,17 +21,16 @@ along with Octave; see the file COPYING.
 */
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #include "error.h"
 #include "oct-hdf5.h"
-#include "oct-hdf5-types.h"
 
 bool
 check_hdf5_types (bool warn)
 {
   static bool checked = false;
   static bool ok = false;
 
   if (! checked)
diff --git a/libinterp/corefcn/oct-hdf5.h b/libinterp/corefcn/oct-hdf5.h
--- a/libinterp/corefcn/oct-hdf5.h
+++ b/libinterp/corefcn/oct-hdf5.h
@@ -18,22 +18,25 @@ You should have received a copy of the G
 along with Octave; see the file COPYING.  If not, see
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if ! defined (octave_oct_hdf5_h)
 #define octave_hdf5_h 1
 
-#if defined (HAVE_HDF5_H)
+#if defined (HAVE_HDF5)
 
+#if defined (HAVE_HDF5_H)
 #include <hdf5.h>
+#endif
 
 #include "oct-hdf5-types.h"
 
 #ifdef USE_64_BIT_IDX_T
 #define H5T_NATIVE_IDX H5T_NATIVE_INT64
 #else
 #define H5T_NATIVE_IDX H5T_NATIVE_INT
 #endif
+
 #endif
 
 #endif
